recipe/build.sh

```diff
@@ -23,12 +11,13 @@ export CPPFLAGS="$CPPFLAGS -DCONDA_PREFIX=\\\"${PREFIX}\\\""
 # we need to redefine that to `python`.
 _PY=$PYTHON
 export PYTHON="python"
+unset _CONDA_PYTHON_SYSCONFIGDATA_NAME
```

Comment:
just wondering why this is added (back)?

Reply:
This comes from conda-forge. I have trouble understanding the whole picture, but I don't think we are affected either way. It seems to have been added around the time setuptools started including distutils. I'll try removing that line.

Reply:
If I had to guess, I would be tempted to say that it might be related to meson not being able to find the right headers or libraries. I personally fear that removing this line could have some sneaky side-effects...

Reply:
https://github.com/conda-forge/glib-feedstock/pull/72 has the history.

Reply:
The compiler activation scripts set something like `_CONDA_PYTHON_SYSCONFIGDATA_NAME=_sysconfigdata_x86_64_conda_cos7_linux_gnu `
Then our python interpreters up to 3.9 are patched to get abi platform and multiarch from both _PYTHON_SYSCONFIGDATA_NAME and _CONDA_PYTHON_SYSCONFIGDATA_NAME. distutils in setuptools >50,<6X were also patched to do so.

There could have been a mess at some point mixing _PYTHON_SYSCONFIGDATA_NAME and _CONDA_PYTHON_SYSCONFIGDATA_NAME in the context of cross-compilation.

And the origin seem to be the support of native compilers.

Reply:
@JeanChristopheMorinPerso I still want to remove this unset call, given our recipe doesn't have cross-compilation parts.

Reply:
I disagree (I like to keep our recipes as close to cf as possible and keep cross-compiling support)), but I can live with it. I already approved the PR :)

Reply:
I thought there were other differences (from not seeing cross stuff in the build script) but actually not. I'll come back to it tomorrow with a fresh mind

@danpetry thoughts?

Reply:
I think it's probably safer to keep the env variable set, but I haven't dug into it/understand enough about it to make a good judgement about it tbh. Probably it's fine? By the look at the linked issue, looks like it was perhaps legacy stuff that's not required after the distutils/setuptools merge, maybe?

Reply:
I reverted the last changes and added a comment for context.